diesel_cli/src/main.rs

```diff
@@ -191,22 +221,23 @@ fn redo_latest_migration<Conn>(conn: &Conn) where
     }).unwrap_or_else(handle_error);
 }
 
-fn handle_error<E: Error>(error: E) {
+fn handle_error<E: Error, T>(error: E) -> T {
```

Comment:
A better way to indicate this is to return `!`, which is the type of `panic!()`, which should have less inference errors and actually indicate that code following this is unreachable.